### PR TITLE
Spanner - add default retryable codes

### PR DESCRIPTION
**Pull request by darshan-sj**
Spanner - add default retryable codes

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by darshan-sj** [comment]
Run Java_GCP_IO_Direct PreCommit

**Review by bvolpato — APPROVED** [review]

**Comment by pabloem** [comment]
Run Java PreCommit

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/SpannerAccessor.java [hunk 1]
```diff
@@ -113,6 +113,9 @@ private static SpannerAccessor createAndConnect(SpannerConfig spannerConfig) {
     SpannerOptions.Builder builder = SpannerOptions.newBuilder();
 
     Set<Code> retryableCodes = new HashSet<>();
+    // Add default retryable codes
+    retryableCodes.addAll(
+        builder.getSpannerStubSettingsBuilder().executeStreamingSqlSettings().getRetryableCodes());
```
Comment (olavloite) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/SpannerAccessor.java:118]: We are taking the retryable codes from the `executeStreamingSql` RPC here, and then apply that to the `executeStreamingSql` __and__ to all unary RPCs below. I don't think we should be doing the latter. Are there any specific unary RPCs that we want to set the retryable codes for? Or do we really want to set them for __all__ of them? E.g. this takes the default retryable codes of `ExecuteStreamingSql` and then applies them to RPCs like `Commit`.

**Review by olavloite — APPROVED** [review]

**Review by rajatbhatta — APPROVED** [review]

**Comment by riteshghorse** [comment]
I'll merge once all tests passes.
Thanks everyone!

**Comment by darshan-sj** [comment]
Run Java PreCommit

**Comment by darshan-sj** [comment]
Run Java PreCommit

**Comment by riteshghorse** [comment]
I think Java PreCommit failure is unrelated to this PR. 
image

**Comment by darshan-sj** [comment]
Yes. The failure is not related to this PR and it is flaky. It had passed a couple of runs back.

**Comment by riteshghorse** [comment]
https://ci-beam.apache.org/job/beam_PreCommit_Java_Cron/6867/ looks like test timed out failure from this build

Can someone from Java take a look,  @Abacn what do you think?

**Comment by Abacn** [comment]
Yes it's known issue, when jenkins gets more heavy load it's timing out more likely. Current practice is go ahead it

**Comment by Abacn** [comment]
btw Java PreCommit even should not trigger in this PR by default as it only touches source under `io/google-cloud-platform`. The test was triggered by phrase

**Comment by pabloem** [comment]
thanks y'all

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/SpannerAccessor.java [hunk 2]
```diff
@@ -113,6 +113,8 @@ private static SpannerAccessor createAndConnect(SpannerConfig spannerConfig) {
     SpannerOptions.Builder builder = SpannerOptions.newBuilder();
 
     Set<Code> retryableCodes = new HashSet<>();
+    // Add default retryable codes
+    retryableCodes.add(Code.UNAVAILABLE);
```
[the remark this answers is not in the document]
Comment (nielm) [reply sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/SpannerAccessor.java]: Resolved by other changes.

Lgtm

**Pull request merged**